admin_guide/managing_networking.adoc

```diff
@@ -758,6 +759,144 @@ You can also specify the `EGRESS_HTTP_PROXY_DESTINATION` using a
 ConfigMap, similarly to
 xref:admin-guide-manage-pods-egress-router-configmap[the redirecting egress router example above].
 
+[[admin-guide-deploying-an-egress-router-dns-proxy-pod]]
+==== Deploying an Egress Router DNS Proxy Pod
+
+In _DNS proxy mode_, the egress router runs as a DNS proxy for TCP based services.
+
+. Create environment variables as ConfigMap object using the following as an example:
++
+[source, yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: egress-router-env
+data:
+  EGRESS_SOURCE: 172.17.0.99 <1>
+  EGRESS_GATEWAY: 172.17.0.1 <2>
+  EGRESS_ROUTER_MODE: dns-proxy <3>
+  EGRESS_DESTINATION: | <4>
+    # Egress routes for Project "Foo", version 5
+
+    80  203.0.113.25
+
+    100 example.com
+
+    8080 203.0.113.26 80
+
+    8443 foobar.com 443
+----
+<1> An IP address from the physical network that the node itself is on and is
+reserved by the cluster administrator for use by this pod.
+<2> Same value as the default gateway used by the node itself.
+<3> This tells the egress router image that it is being deployed as
+part of DNS proxy, and so it should set up dns proxy related iptables
+rules.
+<4> This uses the YAML syntax for a multi-line string; see below for
+details.
++
+Each line of `EGRESS_DESTINATION` can be one of two types:
++
+- `<port> <remote address>` - This says that incoming
+connections to the given `<port>` should be redirected to the same
+TCP port on the given `<remote address>`. `<remote address>` can be an IP address or DNS name.
+In case of DNS name, DNS resolution is done at runtime.
+In the example above, the first line redirects TCP traffic from
+local port 80 to port 80 on 203.0.113.25. The second line redirects TCP traffic from
+local port 100 to port 100 on example.com.
+- `<port> <remote address> <remote port>` - As above, except
+that the connection is redirected to a different `<remote port>` on
+`<remote address>`. In the example above, the third line
+redirects local port 8080 to remote port 80 on 203.0.113.26 and the fourth line
+redirects local port 8443 to remote port 443 on foobar.com.
+
+. Create the pod using the following as an example:
++
+[source, yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: egress-dns-proxy
+  labels:
+    name: egress-dns-proxy
+  annotations:
+    pod.network.openshift.io/assign-macvlan: "true" <1>
+spec:
+  initContainers:
+  - name: egress-router-setup
+ifdef::openshift-enterprise[]
+    image: registry.access.redhat.com/openshift3/ose-egress-router
+endif::openshift-enterprise[]
+ifdef::openshift-origin[]
+    image: openshift/origin-egress-router
+endif::openshift-origin[]
+    securityContext:
+      privileged: true
+    envFrom: <2>
+    - configMapRef:
+        name: egress-router-env
```

Comment:
Blah, just got back from PTO and noticed this. It's not safe to use `envFrom` here, because that might allow project admins to subvert the functioning of the (privileged) origin-egress-router pod by setting variables like `IFS` or `PATH` in the environment that `egress-router.sh` will run in.

Given that the PR implementing this feature hasn't landed yet anyway, maybe this PR should just be reverted?